### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by GaryElshaw** [transmission/transmission#4154] [General]
In the last few months i've been flicking between macOS (my default), Ubuntu and Manjaro. It's been an eye-opener insofar as how the different Transmission clients behave, and also how they perform.

On Manjaro, the 4 beta (including daily builds) performs badly insofar as connectivity and upload and download speeds are concerned. The reason i say that is because i've been testing with other clients. I tried the Fragments client tonight (ostensibly a Transmission 3 client). Not only did i see 26 mbs download for the first time, but there was no corruption, after booting back into mac and seeding.

Three things. 1. V. 4 should of course perform better than V. 3
2. Corrupted downloaded data is an issue i constantly see, often necessitating the use of a different client. Can it be resolved?
3. How can upload and download connectivity/speeds be improved?

**Comment by ckerr** [comment]
I'm a little frustrated, just a few days ago I asked you to not file omnibus tickets, and this is an omnibus ticket.

- Comparing client A to B is a little like saying performance car A is faster or slower than performance car B based on how fast they're moving in rush hour traffic. Nearly all of the limitations are based on the other peers in the swarm / the other cars on the road. Of course I want Transmission to be fast, but if we're going to criticize (or praise!) its performance we need some real metrics.

- IDK how much the "piece X is corrupt" issue happens in real life, but it is tracked in the issue tracker and I have milestoned it for further investigation. Discussing it here just duplicates that effort and, if you have actionable information to report, will make it harder to find that actionable info when doing issue triage.

Since we all want Transmission to be fast and error-free I know I'm going to look like a bad guy locking this discussion. But please discuss the data corruption issue in the data corruption ticket, and please bring **actionable** performance bug reports to a new issue. Thank you!

**Reply by GaryElshaw** [reply]
I don't know why these are popping back up. This is from mid November, the other is from March.

**Reply by ckerr** [reply]
Oog! Yeah you're right, sorry for the mild flaming I gave then. These showed up in my notifications as new and I didn't notice the date. I guess GItHub's backend is having a normal one today. Maybe someone giving it a thumbs-up acts as a bump?

But fr, how dare you post Yet Another Omnibus ticket just a month before I asked you not to :smile:
